install/wipe_ncn_disks_for_reinstallation.md

```diff
@@ -228,34 +262,83 @@ RAIDs, zeroing the disks, and then wiping the disks and RAIDs.
         7741d50966259410298bb4c3210e6665cdbd57a82e34e467d239f519ae3f17d4
         ```
 
-   1. Remove metal LVM.
+1. Remove etcd device **on master nodes ONLY**.
 
-        ```bash
-        ncn# vgremove -f -v --select 'vg_name=~metal*'
-        ```
+   1. This `dmsetup` comand wil determine whether an etcd volume is present.
 
-      > **NOTE:** Optionally, run the `pvs` command. If any drives are still listed, remove them with `pvremove`, but this is rarely needed. Also, if the above command fails or returns a warning about the filesystem being in use, ignore the error and proceed to the next step, as this will not inhibit the wipe process.
+      ```bash
+      ncn-m# dmsetup -ls 
+      ```
 
-1. Stop the RAIDs.
+      Expected output when the etcd volume is present will show `ETCDLVM`, but the numbers might be different.
 
-    ```bash
-    ncn# for md in /dev/md/*; do mdadm -S -v $md || echo nope ; done
-    ```
+      ```bash
+      ETCDLVM (254:1)
+      ```
 
-1. List the disks for verification.
+   1. This `dmsetup` command will remove the etcd device mapper.
 
-    ```bash
-    ncn# ls -1 /dev/sd* /dev/disk/by-label/*
-    ```
+      ```bash
+      ncn-m# dmsetup remove $(dmsetup ls | grep -i etcd | awk '{print $1}')
+      ```
 
-1. Wipe the disks and RAIDs.
+      > **Note:** The following output  means the etcd volume  mapper is not present.
+      ```bash
+      No device specified.
+      Command failed.
+      ```
 
-    ```bash
-    ncn# sgdisk --zap-all /dev/sd*
-    ncn# wipefs --all --force /dev/sd* /dev/disk/by-label/*
-    ```
+1. Remove etcd Volumes **on master nodes ONLY**.
+
+   ```bash
+   ncn-m# vgremove etcdvg0
+   ```
+
+1. Remove metal LVM.
+
+     ```bash
+     ncn# vgremove -f -v --select 'vg_name=~metal*'
+     ```
+
+   > **NOTE:** Optionally, run the `pvs` command. If any drives are still listed, remove them with `pvremove`, but this is rarely needed. Also, if the above command fails or returns a warning about the filesystem being in use, ignore the error and proceed to the next step, as this will not inhibit the wipe process.
+
+1. Group these commands together for each node.
+
+   This group of commands should be done in succession on one node before moving to do the same set of commands on the next node. The nodes would be addressed in descending order for each type of node.  Start with the utility storage nodes, then the worker nodes, then ncn-m003, then ncn-m002. 
+
+   > **WARNING:** Do not run these commands on ncn-m001
+
+   1. Stop the RAIDs.
+
+       ```bash
+       ncn# for md in /dev/md/*; do mdadm -S -v $md || echo nope ; done
+       ```
+
+   1. List the disks for verification.
+
+       ```bash
+       ncn# ls -1 /dev/sd* /dev/disk/by-label/*
+       ```
+
+   1. Wipe the disks and RAIDs.
+
+       ```bash
+       ncn# wipefs --all --force /dev/sd* /dev/disk/by-label/*
```

Comment:
Does this need to be USB safe? This would fetch all non-USB busses that we use in Shasta:
```bash
doomed_disks=$(lsblk -l -o SIZE,NAME,TYPE,TRAN | grep -E '(raid|sata|nvme|sas)' | sort -u | awk '{print "/dev/"$2}' | tr '\n' ' ')
    for doomed_disk in $doomed_disks; do
        wipefs --all --force "$doomed_disk" 2> /dev/null
    done
```

(code borrowed from: https://github.com/Cray-HPE/dracut-metal-mdsquash/blob/main/90metalmdsquash/metal-md-lib.sh#L298)

Reply:
We already have a warning not to run this command on ncn-m001 which would be the only node with a USB device, unless the customer has done something non-standard.